crates/pop-cli/src/commands/up/contract.rs

```diff
@@ -140,4 +147,41 @@ impl UpContractCommand {
 		}
 		Ok(())
 	}
+
+	/// Uploads only the contract without instantiating it.
+	async fn upload_contract(self) -> anyhow::Result<()> {
+		let upload_exec = set_up_upload(UpOpts {
```

Comment:
Can you implement `From<UpContractCommand> for UpOpts` so that this can just become `self.into()` ?

Reply:
Should `pop_parachains` crate know about this type: `UpContractCommand`?

Reply:
Noted for a future refactor.

Reply:
Implemented